### PR TITLE
enable selective chart publishing via tags

### PR DESCRIPTION
**Pull request by krancour**
This PR would disable auto-publishing of all charts in this GH repo to the chart repo upon merge to master. In practice, that has been messy because it requires that _every_ PR revs the version in the `chart.yaml` of any chart it touches and we haven't been rigorous about enforcing that. It is the reason, for instance, that the brigade 1.0.0 chart is a little different today than it was yesterday. Users expect things tagged with a semantic version to be immutable.

What this will do, instead, is allow us to be a bit more deliberate in chart releasing / publishing. For any given chart, we have the opportunity to do _many_ merges to master, accumulating a series of changes, then, when ready, revving the version number and tagging to release.

__EDIT:__ Two recent additions to this after a bunch of testing proved they were necessary:

1. I set the version of _all_ `Chart.yaml` files to `0.0.1` (it needs to be a valid semantic version if you ever want to `helm install` straight from the local file system. The _real_ value is discerned from the tag and injected at release time using `helm package --version ...`. This absolves maintainers from the burden of manually revving the version in any given chart's `Chart.yaml` just prior to its release.

1. I found a bug (also exists in master) where we were building an index by merging the existing `index.yaml` from the `master` branch with the newly computed index and then publishing that to the chart repo. But the `index.yaml` in master _never_ go updated. So _every_ time a release occurs, new stuff gets added to very old stuff and anything in between gets lost! There hasn't been enough activity in this repo to really notice this, but it is for sure happening. I corrected this bug by amending the indexing process to fetch the latest index from the `gh-pages` branch and merge new stuff with _that_. There no longer needs to be an `index.yaml` anywhere other than the `gh-pages` branch, so I am now .gitignoring the entire `dist/` dir.

### CODE REVIEW

**Review by vdice — APPROVED** [review]
This is great!  Definitely thumbs up on the approach.  I haven't tested this manually, but I can help do so if you'd like...

**Comment by krancour** [comment]
@vdice I had trouble testing this e2e, so that would be great. I did, however, verify the function of the new logic with a small test program.

**Review by vdice — COMMENTED** [review]
This all looks great.  Glad the additional issues noticed have been addressed as well.

My only additional request would be adding some docs on the revised release process, i.e., example(s) showing what tag format one would use for tagging a commit and what occurs as a result courtesy the Brigade pipeline, perhaps also mentioning the revised chart versioning scheme in general.  I suppose this could be as simple as a new section in our very minimal README.md.  

On a related note, we should update Brigade's [releasing doc](https://github.com/brigadecore/brigade/blob/master/docs/content/topics/releasing.md), at least where it pertains cutting chart versions, in a follow-up PR.

**Comment by krancour** [comment]
@vdice a very reasonable ask. Will update docs tomorrow. Thanks for reviewing.

File: README.md [hunk 1]
```diff
@@ -5,10 +5,87 @@ Helm charts for the [Brigade](https://github.com/brigadecore/brigade) project.
...
+for example:
+
+```
+$ git checkout master
+$ git pull upstream master
+$ git tag kashti-v1.0.0
+$ git push upstream --tags
+```
+
+Note that only a project maintainer is able to performt the steps above.
```
Comment (vdice) [inline README.md]: typo: `performt`

**Review by vdice — APPROVED** [review]
Added docs are 💯 !  One lil typo; otherwise LGTM.

Thanks for spending the time to address some major issues here and overhaul the process for the better.

File: README.md (hunk 1 above)
Comment (krancour) [reply README.md]: You have a great eye for typos! Fixing.

Comment (krancour) [reply README.md]: Fixed.

**Pull request merged**